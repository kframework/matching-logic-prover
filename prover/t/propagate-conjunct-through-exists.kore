axiom a: \or(\exists{A{Int}} \and(A{Int}), \exists{X{Int}} \and(gte(X{Int}, 2), \exists{Y{Int}} gte(Y{Int}, 1)))

claim \or(\exists{A{Int}} \and(A{Int}), \exists{X{Int},Y{Int}} \and(gte(Y{Int}, 1), gte(X{Int}, 2)))
strategy propagate-conjunct-through-exists(1, 1)
       . apply(a, fail)
