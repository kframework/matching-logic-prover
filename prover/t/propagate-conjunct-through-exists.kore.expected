<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom a : \or ( \exists { A { Int } , .Patterns } \and ( A { Int } , .Patterns ) , \exists { X { Int } , .Patterns } \and ( gte ( X { Int } , 2 , .Patterns ) , \exists { Y { Int } , .Patterns } gte ( Y { Int } , 1 , .Patterns ) , .Patterns ) , .Patterns )
    </declaration> <declaration>
      axiom ax0 : \or ( \exists { A { Int } , .Patterns } \and ( A { Int } , .Patterns ) , \exists { X { Int } , Y { Int } , .Patterns } \and ( gte ( Y { Int } , 1 , .Patterns ) , gte ( X { Int } , 2 , .Patterns ) , .Patterns ) , .Patterns )
    </declaration>
  </declarations>
</prover>
